File: clients/typescript/src/apis/MediaApi.ts
/* tslint:disable */
/* eslint-disable */

import * as runtime from '../runtime';
import {
    type HTTPValidationError,
    HTTPValidationErrorFromJSON,
    HTTPValidationErrorToJSON,
} from '../models/HTTPValidationError';
import {
    type MediaFormats,
    MediaFormatsFromJSON,
    MediaFormatsToJSON,
} from '../models/MediaFormats';
import {
    type MediaFrame,
    MediaFrameFromJSON,
    MediaFrameToJSON,
} from '../models/MediaFrame';
import {
    type MediaInfo,
    MediaInfoFromJSON,
    MediaInfoToJSON,
} from '../models/MediaInfo';
import {
    type TranscriptSegment,
    TranscriptSegmentFromJSON,
    TranscriptSegmentToJSON,
} from '../models/TranscriptSegment';

export interface GetMediaBytesRequest {

    blobId: number;

    maxDim?: number;

    waveform?: number;
}

export interface GetMediaInfoRequest {

    blobId: number;

    customerId?: string | null;
}

export interface ListMediaFramesRequest {

    blobId: number;

    customerId?: string | null;
}

export interface ListMediaTranscriptRequest {

    blobId: number;

    customerId?: string | null;
}

export class MediaApi extends runtime.BaseAPI {

    async getMediaBytesRequestOpts(requestParameters: GetMediaBytesRequest): Promise<runtime.RequestOpts> {
        if (requestParameters['blobId'] == null) {
            throw new runtime.RequiredError(
                'blobId',
                'Required parameter "blobId" was null or undefined when calling getMediaBytes().'
            );
        }

        const queryParameters: any = {};

        if (requestParameters['maxDim'] != null) {
            queryParameters['max_dim'] = requestParameters['maxDim'];
        }

        if (requestParameters['waveform'] != null) {
            queryParameters['waveform'] = requestParameters['waveform'];
        }

        const headerParameters: runtime.HTTPHeaders = {};

        if (this.configuration && this.configuration.accessToken) {
            const token = this.configuration.accessToken;
            const tokenString = await token("bearerAuth", []);

            if (tokenString) {
                headerParameters["Authorization"] = `Bearer ${tokenString}`;
            }
        }

        let urlPath = `/v1/media/{blob_id}/bytes`;
        urlPath = urlPath.replace('{blob_id}', encodeURIComponent(String(requestParameters['blobId'])));

        return {
            path: urlPath,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        };
    }

    async getMediaBytesRaw(requestParameters: GetMediaBytesRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<any>> {
        const requestOptions = await this.getMediaBytesRequestOpts(requestParameters);
        const response = await this.request(requestOptions, initOverrides);

        if (this.isJsonMime(response.headers.get('content-type'))) {
            return new runtime.JSONApiResponse<any>(response);
        } else {
            return new runtime.TextApiResponse(response) as any;
        }
    }

    async getMediaBytes(requestParameters: GetMediaBytesRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<any> {
        const response = await this.getMediaBytesRaw(requestParameters, initOverrides);
        return await response.value();
    }

    async getMediaInfoRequestOpts(requestParameters: GetMediaInfoRequest): Promise<runtime.RequestOpts> {
        if (requestParameters['blobId'] == null) {
            throw new runtime.RequiredError(
                'blobId',
                'Required parameter "blobId" was null or undefined when calling getMediaInfo().'
            );
        }

        const queryParameters: any = {};

        if (requestParameters['customerId'] != null) {
            queryParameters['customer_id'] = requestParameters['customerId'];
        }

        const headerParameters: runtime.HTTPHeaders = {};

        if (this.configuration && this.configuration.accessToken) {
            const token = this.configuration.accessToken;
            const tokenString = await token("bearerAuth", []);

            if (tokenString) {
                headerParameters["Authorization"] = `Bearer ${tokenString}`;
            }
        }

        let urlPath = `/v1/media/{blob_id}`;
        urlPath = urlPath.replace('{blob_id}', encodeURIComponent(String(requestParameters['blobId'])));

        return {
            path: urlPath,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        };
    }

    async getMediaInfoRaw(requestParameters: GetMediaInfoRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<MediaInfo>> {
        const requestOptions = await this.getMediaInfoRequestOpts(requestParameters);
        const response = await this.request(requestOptions, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => MediaInfoFromJSON(jsonValue));
    }

    async getMediaInfo(requestParameters: GetMediaInfoRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<MediaInfo> {
        const response = await this.getMediaInfoRaw(requestParameters, initOverrides);
        return await response.value();
    }

    async listMediaFormatsRequestOpts(): Promise<runtime.RequestOpts> {
        const queryParameters: any = {};

        const headerParameters: runtime.HTTPHeaders = {};

        if (this.configuration && this.configuration.accessToken) {
            const token = this.configuration.accessToken;
            const tokenString = await token("bearerAuth", []);

            if (tokenString) {
                headerParameters["Authorization"] = `Bearer ${tokenString}`;
            }
        }

        let urlPath = `/v1/media/formats`;

        return {
            path: urlPath,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        };
    }

    async listMediaFormatsRaw(initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<MediaFormats>> {
        const requestOptions = await this.listMediaFormatsRequestOpts();
        const response = await this.request(requestOptions, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => MediaFormatsFromJSON(jsonValue));
    }

    async listMediaFormats(initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<MediaFormats> {
        const response = await this.listMediaFormatsRaw(initOverrides);
        return await response.value();
    }

    async listMediaFramesRequestOpts(requestParameters: ListMediaFramesRequest): Promise<runtime.RequestOpts> {
        if (requestParameters['blobId'] == null) {
            throw new runtime.RequiredError(
                'blobId',
                'Required parameter "blobId" was null or undefined when calling listMediaFrames().'
            );
        }

        const queryParameters: any = {};

        if (requestParameters['customerId'] != null) {
            queryParameters['customer_id'] = requestParameters['customerId'];
        }

        const headerParameters: runtime.HTTPHeaders = {};

        if (this.configuration && this.configuration.accessToken) {
            const token = this.configuration.accessToken;
            const tokenString = await token("bearerAuth", []);

            if (tokenString) {
                headerParameters["Authorization"] = `Bearer ${tokenString}`;
            }
        }

        let urlPath = `/v1/media/{blob_id}/frames`;
        urlPath = urlPath.replace('{blob_id}', encodeURIComponent(String(requestParameters['blobId'])));

        return {
            path: urlPath,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        };
    }

    async listMediaFramesRaw(requestParameters: ListMediaFramesRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<Array<MediaFrame>>> {
        const requestOptions = await this.listMediaFramesRequestOpts(requestParameters);
        const response = await this.request(requestOptions, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(MediaFrameFromJSON));
    }

    async listMediaFrames(requestParameters: ListMediaFramesRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<Array<MediaFrame>> {
        const response = await this.listMediaFramesRaw(requestParameters, initOverrides);
        return await response.value();
    }

    async listMediaTranscriptRequestOpts(requestParameters: ListMediaTranscriptRequest): Promise<runtime.RequestOpts> {
        if (requestParameters['blobId'] == null) {
            throw new runtime.RequiredError(
                'blobId',
                'Required parameter "blobId" was null or undefined when calling listMediaTranscript().'
            );
        }

        const queryParameters: any = {};

        if (requestParameters['customerId'] != null) {
            queryParameters['customer_id'] = requestParameters['customerId'];
        }

        const headerParameters: runtime.HTTPHeaders = {};

        if (this.configuration && this.configuration.accessToken) {
            const token = this.configuration.accessToken;
            const tokenString = await token("bearerAuth", []);

            if (tokenString) {
                headerParameters["Authorization"] = `Bearer ${tokenString}`;
            }
        }

        let urlPath = `/v1/media/{blob_id}/transcript`;
        urlPath = urlPath.replace('{blob_id}', encodeURIComponent(String(requestParameters['blobId'])));

        return {
            path: urlPath,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        };
    }

    async listMediaTranscriptRaw(requestParameters: ListMediaTranscriptRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<runtime.ApiResponse<Array<TranscriptSegment>>> {
        const requestOptions = await this.listMediaTranscriptRequestOpts(requestParameters);
        const response = await this.request(requestOptions, initOverrides);

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(TranscriptSegmentFromJSON));
    }

    async listMediaTranscript(requestParameters: ListMediaTranscriptRequest, initOverrides?: RequestInit | runtime.InitOverrideFunction): Promise<Array<TranscriptSegment>> {
        const response = await this.listMediaTranscriptRaw(requestParameters, initOverrides);
        return await response.value();
    }

}

File: clients/rust/src/apis/media_api.rs
use reqwest;
use serde::{Deserialize, Serialize, de::Error as _};
use crate::{apis::ResponseContent, models};
use super::{Error, configuration, ContentType};


#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum GetMediaBytesError {
    Status422(models::HttpValidationError),
    UnknownValue(serde_json::Value),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum GetMediaInfoError {
    Status422(models::HttpValidationError),
    UnknownValue(serde_json::Value),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum ListMediaFormatsError {
    UnknownValue(serde_json::Value),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum ListMediaFramesError {
    Status422(models::HttpValidationError),
    UnknownValue(serde_json::Value),
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(untagged)]
pub enum ListMediaTranscriptError {
    Status422(models::HttpValidationError),
    UnknownValue(serde_json::Value),
}


pub async fn get_media_bytes(configuration: &configuration::Configuration, blob_id: i32, max_dim: Option<i32>, waveform: Option<i32>) -> Result<serde_json::Value, Error<GetMediaBytesError>> {
    let p_path_blob_id = blob_id;
    let p_query_max_dim = max_dim;
    let p_query_waveform = waveform;

    let uri_str = format!("{}/v1/media/{blob_id}/bytes", configuration.base_path, blob_id=p_path_blob_id);
    let mut req_builder = configuration.client.request(reqwest::Method::GET, &uri_str);

    if let Some(ref param_value) = p_query_max_dim {
        req_builder = req_builder.query(&[("max_dim", &param_value.to_string())]);
    }
    if let Some(ref param_value) = p_query_waveform {
        req_builder = req_builder.query(&[("waveform", &param_value.to_string())]);
    }
    if let Some(ref user_agent) = configuration.user_agent {
        req_builder = req_builder.header(reqwest::header::USER_AGENT, user_agent.clone());
    }
    if let Some(ref token) = configuration.bearer_access_token {
        req_builder = req_builder.bearer_auth(token.to_owned());
    };

    let req = req_builder.build()?;
    let resp = configuration.client.execute(req).await?;

    let status = resp.status();
    let content_type = resp
        .headers()
        .get("content-type")
        .and_then(|v| v.to_str().ok())
        .unwrap_or("application/octet-stream");
    let content_type = super::ContentType::from(content_type);

    if !status.is_client_error() && !status.is_server_error() {
        let content = resp.text().await?;
        match content_type {
            ContentType::Json => serde_json::from_str(&content).map_err(Error::from),
            ContentType::Text => return Err(Error::from(serde_json::Error::custom("Received `text/plain` content type response that cannot be converted to `serde_json::Value`"))),
            ContentType::Unsupported(unknown_type) => return Err(Error::from(serde_json::Error::custom(format!("Received `{unknown_type}` content type response that cannot be converted to `serde_json::Value`")))),
        }
    } else {
        let content = resp.text().await?;
        let entity: Option<GetMediaBytesError> = serde_json::from_str(&content).ok();
        Err(Error::ResponseError(ResponseContent { status, content, entity }))
    }
}

pub async fn get_media_info(configuration: &configuration::Configuration, blob_id: i32, customer_id: Option<&str>) -> Result<models::MediaInfo, Error<GetMediaInfoError>> {
    let p_path_blob_id = blob_id;
    let p_query_customer_id = customer_id;

    let uri_str = format!("{}/v1/media/{blob_id}", configuration.base_path, blob_id=p_path_blob_id);
    let mut req_builder = configuration.client.request(reqwest::Method::GET, &uri_str);

    if let Some(ref param_value) = p_query_customer_id {
        req_builder = req_builder.query(&[("customer_id", &param_value.to_string())]);
    }
    if let Some(ref user_agent) = configuration.user_agent {
        req_builder = req_builder.header(reqwest::header::USER_AGENT, user_agent.clone());
    }
    if let Some(ref token) = configuration.bearer_access_token {
        req_builder = req_builder.bearer_auth(token.to_owned());
    };

    let req = req_builder.build()?;
    let resp = configuration.client.execute(req).await?;

    let status = resp.status();
    let content_type = resp
        .headers()
        .get("content-type")
        .and_then(|v| v.to_str().ok())
        .unwrap_or("application/octet-stream");
    let content_type = super::ContentType::from(content_type);

    if !status.is_client_error() && !status.is_server_error() {
        let content = resp.text().await?;
        match content_type {
            ContentType::Json => serde_json::from_str(&content).map_err(Error::from),
            ContentType::Text => return Err(Error::from(serde_json::Error::custom("Received `text/plain` content type response that cannot be converted to `models::MediaInfo`"))),
            ContentType::Unsupported(unknown_type) => return Err(Error::from(serde_json::Error::custom(format!("Received `{unknown_type}` content type response that cannot be converted to `models::MediaInfo`")))),
        }
    } else {
        let content = resp.text().await?;
        let entity: Option<GetMediaInfoError> = serde_json::from_str(&content).ok();
        Err(Error::ResponseError(ResponseContent { status, content, entity }))
    }
}

pub async fn list_media_formats(configuration: &configuration::Configuration, ) -> Result<models::MediaFormats, Error<ListMediaFormatsError>> {

    let uri_str = format!("{}/v1/media/formats", configuration.base_path);
    let mut req_builder = configuration.client.request(reqwest::Method::GET, &uri_str);

    if let Some(ref user_agent) = configuration.user_agent {
        req_builder = req_builder.header(reqwest::header::USER_AGENT, user_agent.clone());
    }
    if let Some(ref token) = configuration.bearer_access_token {
        req_builder = req_builder.bearer_auth(token.to_owned());
    };

    let req = req_builder.build()?;
    let resp = configuration.client.execute(req).await?;

    let status = resp.status();
    let content_type = resp
        .headers()
        .get("content-type")
        .and_then(|v| v.to_str().ok())
        .unwrap_or("application/octet-stream");
    let content_type = super::ContentType::from(content_type);

    if !status.is_client_error() && !status.is_server_error() {
        let content = resp.text().await?;
        match content_type {
            ContentType::Json => serde_json::from_str(&content).map_err(Error::from),
            ContentType::Text => return Err(Error::from(serde_json::Error::custom("Received `text/plain` content type response that cannot be converted to `models::MediaFormats`"))),
            ContentType::Unsupported(unknown_type) => return Err(Error::from(serde_json::Error::custom(format!("Received `{unknown_type}` content type response that cannot be converted to `models::MediaFormats`")))),
        }
    } else {
        let content = resp.text().await?;
        let entity: Option<ListMediaFormatsError> = serde_json::from_str(&content).ok();
        Err(Error::ResponseError(ResponseContent { status, content, entity }))
    }
}

pub async fn list_media_frames(configuration: &configuration::Configuration, blob_id: i32, customer_id: Option<&str>) -> Result<Vec<models::MediaFrame>, Error<ListMediaFramesError>> {
    let p_path_blob_id = blob_id;
    let p_query_customer_id = customer_id;

    let uri_str = format!("{}/v1/media/{blob_id}/frames", configuration.base_path, blob_id=p_path_blob_id);
    let mut req_builder = configuration.client.request(reqwest::Method::GET, &uri_str);

    if let Some(ref param_value) = p_query_customer_id {
        req_builder = req_builder.query(&[("customer_id", &param_value.to_string())]);
    }
    if let Some(ref user_agent) = configuration.user_agent {
        req_builder = req_builder.header(reqwest::header::USER_AGENT, user_agent.clone());
    }
    if let Some(ref token) = configuration.bearer_access_token {
        req_builder = req_builder.bearer_auth(token.to_owned());
    };

    let req = req_builder.build()?;
    let resp = configuration.client.execute(req).await?;

    let status = resp.status();
    let content_type = resp
        .headers()
        .get("content-type")
        .and_then(|v| v.to_str().ok())
        .unwrap_or("application/octet-stream");
    let content_type = super::ContentType::from(content_type);

    if !status.is_client_error() && !status.is_server_error() {
        let content = resp.text().await?;
        match content_type {
            ContentType::Json => serde_json::from_str(&content).map_err(Error::from),
            ContentType::Text => return Err(Error::from(serde_json::Error::custom("Received `text/plain` content type response that cannot be converted to `Vec&lt;models::MediaFrame&gt;`"))),
            ContentType::Unsupported(unknown_type) => return Err(Error::from(serde_json::Error::custom(format!("Received `{unknown_type}` content type response that cannot be converted to `Vec&lt;models::MediaFrame&gt;`")))),
        }
    } else {
        let content = resp.text().await?;
        let entity: Option<ListMediaFramesError> = serde_json::from_str(&content).ok();
        Err(Error::ResponseError(ResponseContent { status, content, entity }))
    }
}

pub async fn list_media_transcript(configuration: &configuration::Configuration, blob_id: i32, customer_id: Option<&str>) -> Result<Vec<models::TranscriptSegment>, Error<ListMediaTranscriptError>> {
    let p_path_blob_id = blob_id;
    let p_query_customer_id = customer_id;

    let uri_str = format!("{}/v1/media/{blob_id}/transcript", configuration.base_path, blob_id=p_path_blob_id);
    let mut req_builder = configuration.client.request(reqwest::Method::GET, &uri_str);

    if let Some(ref param_value) = p_query_customer_id {
        req_builder = req_builder.query(&[("customer_id", &param_value.to_string())]);
    }
    if let Some(ref user_agent) = configuration.user_agent {
        req_builder = req_builder.header(reqwest::header::USER_AGENT, user_agent.clone());
    }
    if let Some(ref token) = configuration.bearer_access_token {
        req_builder = req_builder.bearer_auth(token.to_owned());
    };

    let req = req_builder.build()?;
    let resp = configuration.client.execute(req).await?;

    let status = resp.status();
    let content_type = resp
        .headers()
        .get("content-type")
        .and_then(|v| v.to_str().ok())
        .unwrap_or("application/octet-stream");
    let content_type = super::ContentType::from(content_type);

    if !status.is_client_error() && !status.is_server_error() {
        let content = resp.text().await?;
        match content_type {
            ContentType::Json => serde_json::from_str(&content).map_err(Error::from),
            ContentType::Text => return Err(Error::from(serde_json::Error::custom("Received `text/plain` content type response that cannot be converted to `Vec&lt;models::TranscriptSegment&gt;`"))),
            ContentType::Unsupported(unknown_type) => return Err(Error::from(serde_json::Error::custom(format!("Received `{unknown_type}` content type response that cannot be converted to `Vec&lt;models::TranscriptSegment&gt;`")))),
        }
    } else {
        let content = resp.text().await?;
        let entity: Option<ListMediaTranscriptError> = serde_json::from_str(&content).ok();
        Err(Error::ResponseError(ResponseContent { status, content, entity }))
    }
}

